proto/Makefile

```diff
@@ -39,8 +39,15 @@ PROTOC_GEN_GOGO := $(PWD)/../../../../../bin/protoc-gen-gogo
 # (and these directories must be in the include path), but cgo can only compile
 # a single directory so we symlink the generated pb.cc files into the storage/engine
 # directory.
+#
+# The gogoprotobuf customtype and unmarshaller extensions do not play nicely with integer
+# types. Manually wack the generate code to make it compile.
 all:
 	(cd ../.. && $(PROTOC) --plugin=protoc-gen-gogo=$(PROTOC_GEN_GOGO) --gogo_out=. --cpp_out=cockroach/storage/engine --proto_path=.:$(PROTO_PATH) $(PROTOS:%=cockroach/proto/%))
 	(cd ../.. && $(PROTOC) --cpp_out=cockroach/storage/engine --proto_path=$(PROTO_PATH) $(GOGO_PROTOS))
 	(cd ../storage/engine && ln -sf gogoproto/gogo.pb.cc .)
 	(cd ../storage/engine && ln -sf cockroach/proto/*.pb.cc .)
+	sed -E 's/(Node|Store)ID \|= \(int32/\1ID |= (\1ID/g' < config.pb.go > config.pb.go.new && \
```

Comment:
Interesting; based on Bram's experience with `Key` I thought custom types were best supported with scalar types (it uses a pointer to a `proto.Key` which is a byte slice, so there are two different levels which can be nil and this confuses it). Maybe the answer is that they're not very well supported in any case. I wouldn't miss them too much if we stopped using them. Conversions between the various ID integer types have become so habitual that I don't know if they're helping prevent bugs any more.